src/hypothesis/internal/reflection.py

```diff
@@ -45,6 +46,21 @@ def fully_qualified_name(f):
         return qualname(f)
 
 
+def is_mock(obj):
+    """Determine if the given argument is a mock type.
+
+    We want to be able to detect these when dealing with various test
+    args. As they are sneaky and can look like almost anything else,
+    we'll check this by looking for random attributes.  This is more
+    robust than looking for types.
+
+    """
+    if obj is None:
+        return False
+    random_attributes = [uuid.uuid4()] * 10
+    return all([getattr(obj, str(attr), None) for attr in random_attributes])
```

Comment:
For the same effect, but without the special case for None and generating as few UUIDs as possible:

```
for _ in range(10):  # or 20, etc
    if not hasattr(obj, str(uuid.uuid4())):
        return False
return True
```